lib/command/run.rb

```diff
@@ -442,16 +442,22 @@ def print_detached_commands
       )
     end
 
-    def resolve_job_status
-      result = cp.fetch_cron_workload(runner_workload, location: location)
-      job_details = result&.dig("items")&.find { |item| item["id"] == job }
-      status = job_details&.dig("status")
+    def resolve_job_status # rubocop:disable Metrics/MethodLength
+      loop do
+        result = cp.fetch_cron_workload(runner_workload, location: location)
```

Comment:
This could return nil for 1,000 reasons. Can we make this more robust?

Should we generally use the version of `perform_yaml!` that exits if non-zero?

In general we should crash immediately for unexpected output.

If we expect that perform_yaml might have a non-zero exit code, then check std_err for the expected error and otherwise abort.